### PR TITLE
Pull base image by arch

### PR DESCRIPTION
**Pull request by n063h**
### Motivation and Context

We got problems with the existing code：
1. now the workflow build and push images for each platform seperately, so the later uploaded image will overwrite the earlier uploaded image with the same tag.
2. depend on a large and outdated os base image `manylinux2014`.
3. slow

### Description
- use cross compile to accelerate compilation.
- use strategy.matrix to split target platforms to seperate jobs to compile parallelly.
- use buildx --platform to annotate image's platform on dockerhub which avoids overwritting same-tag images.

### How Has This Been Tested?
build envd image with new horust images on amd64/arm64 seperately.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -22,34 +22,59 @@ jobs:
...
+        uses: actions/download-artifact@v3
+        with:
+          name: aarch64-unknown-linux-gnu
+          path: arm64
+      - name: Download linux/amd64 artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: x86_64-unknown-linux-gnu
+          path: amd64
+      - run: chmod 777 arm64/horust &&  chmod 777 amd64/horust
```
Comment (kemingy) [inline .github/workflows/release.yml]: read + exec should be enough?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -22,9 +22,33 @@ jobs:
       - name: Run tests
         run: cargo test --verbose
 
-  docker:
+  compile:
     needs: [test]
-    if: github.repository == 'tensorchord/Horust'
+    # if: github.repository == 'tensorchord/Horust'
```
Comment (kemingy) [inline .github/workflows/release.yml]: To avoid accident release by forked repo.
```suggestion
    if: github.repository == 'tensorchord/Horust'
```

Comment (n063h) [reply .github/workflows/release.yml]: > set 755 permission(test)

Just for a test, I'm releasing it to my dockerhub to check if it works.

Comment (n063h) [reply .github/workflows/release.yml]: I will reset the git history to overwrite this test commit.

Comment (n063h) [reply .github/workflows/release.yml]: Maybe it's better to test on another branch checkouted from this one.

File: .github/workflows/release.yml (hunk 1 above)
Comment (n063h) [reply .github/workflows/release.yml]: yes

**Review by kemingy — APPROVED** [review]

**Pull request merged**